### PR TITLE
Call PlayerMoveEvent.

### PR DESCRIPTION
**Pull request by JeromSar**
Currently, PlayerMoveEvent is not called. This Pull Request adds functionality to address this.

By carefully inspecting CraftBukkit and NMS code, The right packet is called when the event is cancelled: _0x08 Player Rotation_ (previously known as _0x11 Player Position_).

Additionally, this pull request includes a field name refractor in PlayerRotationMessage: from _rotation_ to _yaw_.

See https://docs.google.com/spreadsheet/ccc?key=0Ai7FZY3N9uoYdFZEaU1RZUswZ2dqUFRpTTkyUEk1dXc&usp=sharing#gid=0
And https://github.com/JeromSar/CraftBukkit/blob/master/src/main/java/net/minecraft/server/PlayerConnection.java#L240

### CODE REVIEW

File: src/main/java/net/glowstone/net/handler/play/player/PlayerUpdateHandler.java [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-        if (newLoc.distanceSquared(original) > 16 * 16) {
+        if (newLoc.distanceSquared(originLoc) > 16 * 16) {
+            return;
+        }
+
+        final PlayerMoveEvent event = EventFactory.onPlayerMove(session.getPlayer(), originLoc, newLoc);
+        if (event.isCancelled()) {
+            // Magic value
+            // https://github.com/JeromSar/CraftBukkit/blob/master/src/main/java/net/minecraft/server/PlayerConnection.java#L240
+            session.send(new PositionRotationMessage(originLoc, 1.6200000047683716D)); // Magic value:
```
Comment (turt2live) [inline src/main/java/net/glowstone/net/handler/play/player/PlayerUpdateHandler.java]: `// Magic value:`

._.

Comment (JeromSar) [inline src/main/java/net/glowstone/net/handler/play/player/PlayerUpdateHandler.java]: @turt2live You're saying _1.6200000047683716D_ isn't a magic value?

Comment (JeromSar) [inline src/main/java/net/glowstone/net/handler/play/player/PlayerUpdateHandler.java]: Oh wait, it's doubled, nevermind.

**Comment by JeromSar** [comment]
@turt2live Fixed formatting concerns

**Comment by turt2live** [comment]
@JeromSar Mind publishing your testing materials for this?

**Comment by JeromSar** [comment]
@turt2live Moved this PR to #277. Will publish testing materials there when ready. :)

**Pull request closed without merging**